MatrixSDK/VoIP/MXCallManager.m

```diff
@@ -565,6 +632,54 @@ - (void)handleCallRejectReplacement:(MXEvent *)event
     }
 }
 
+- (void)handleRoomMember:(MXEvent *)event
+{
+    MXRoomMemberEventContent *content = [MXRoomMemberEventContent modelFromJSON:event.content];
+    
+    if ([MXTools membership:content.membership] == MXMembershipInvite)
+    {
+        //  a room invite
+        MXRoom *room = [_mxSession roomWithRoomId:event.roomId];
+        NSString *directUserId = room.directUserId;
+        
+        if (!directUserId && content.isDirect)
+        {
+            directUserId = event.sender;
+        }
+        
+        if (directUserId)
+        {
+            [self getNativeUserFrom:directUserId success:^(MXThirdPartyUserInstance * _Nonnull user) {
+                MXRoom *nativeRoom = [self.mxSession directJoinedRoomWithUserId:user.userId];
+                if (nativeRoom)
+                {
+                    //  auto-accept this invite from the virtual room, if a direct room with the native user found
+                    [room join:^{
+                        NSLog(@"[MXCallManager] handleRoomMember: auto-joined on virtual room successfully.");
+                        
+                        room.summary.hiddenFromUser = YES;
```

Comment:
Can we let `MXRoomSummaryUpdater.checkRoomIsVirtualWithCreateEvent()` do all this?

I am not really confortable with this game between create event params and account data. It looks very hacky and fragile. I would prefer to have it in a single place instead of 4 different places. It makes it very hard to maintain and easy to break.

Reply:
I've introduced  `MXRoomAccountDataUpdater`. It's not perfect but seems the best way to me, as i don't want to include virtual room logic into the room summary updater too much.